.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,41 @@
+<!--
+Thanks for submitting a pull request 🎉! Here are some tips for you:
+
+* If this is your first contribution, read "Cargo Contribution Guide":
+  https://doc.crates.io/contrib/
+* Run `cargo fmt --all` to format your code changes.
+* Small commits and pull requests are always preferable and easy to review.
+* If your idea is large and needs feedback from the community, read how:
+  https://doc.crates.io/contrib/process/#working-on-large-features
+* Cargo takes care of compatibility. Read our design principles:
+  https://doc.crates.io/contrib/design.html
+* When changing help text of cargo commands, follow the steps to generate docs:
+  https://github.com/rust-lang/cargo/tree/master/src/doc#building-the-man-pages
+* If your PR is not finished, set it as "draft" PR or add "WIP" in its title.
+* It's ok to use the CI resources to test your PR, but please don't abuse them.
+
+Please DELETE all these comments before submitting the PR.
+-->
+
+### What this PR tries to resolve?
```

Comment:
Personally I find that automatically-generated-sections are sometimes a little heavy-handed for filling out with a PR. I find that most of the time these end up getting filled with trivial information that doesn't go beyond the title of the PR or in general folks don't really put much effort into filling this out.

I'd ideally like to start with an all-comment template to start out with where if someone ignores it then it's ok in the sense that it doesn't show up anywhere else.

Reply:
That being said I think the comments in these sections are still great to call out so I wouldn't want their contents deleted.